ydb/core/kqp/executer_actor/kqp_planner.cpp

```diff
@@ -832,6 +864,31 @@ void TKqpPlanner::CollectTaskChannelsUpdates(const TKqpTasksGraph::TTaskType& ta
     }
 }
 
+void TKqpPlanner::SendReadyStateToCheckpointCoordinator() {
+    if (CheckpointsReadyStateSended) {
+        return;
+    }
+    auto event = std::make_unique<NFq::TEvCheckpointCoordinator::TEvReadyState>();
+    for (const auto& dqTask : TasksGraph.GetTasks()) {
+        NYql::NDqProto::TDqTask* taskDesc = ArenaSerializeTaskToProto(TasksGraph, dqTask, true);
+        auto settings = NDq::TDqTaskSettings(taskDesc, TasksGraph.GetMeta().GetArenaIntrusivePtr());
+        bool enabledCheckpoints = NYql::NDq::GetTaskCheckpointingMode(settings) != NYql::NDqProto::CHECKPOINTING_MODE_DISABLED;
+        bool isIngress = TasksGraph.IsIngress(dqTask);
+        auto task = NFq::TEvCheckpointCoordinator::TEvReadyState::TTask{
+            dqTask.Id,
+            enabledCheckpoints,
+            isIngress,
+            TasksGraph.IsEgressTask(dqTask),
+            NYql::NDq::HasState(settings),
+            dqTask.ComputeActorId
+        };
+        event->Tasks.emplace_back(std::move(task));
+    }
+    LOG_I("Sending TEvReadyState to checkpoint coordinator (" << CheckpointCoordinatorId << ")");
+    TlsActivationContext->Send(std::make_unique<NActors::IEventHandle>(CheckpointCoordinatorId, ExecuterId, event.release()));
+    CheckpointsReadyStateSended = true;
```

Comment:
Typo `CheckpointsReadyStateSended` -> `CheckpointsReadyStateSent`

```suggestion
    CheckpointsReadyStateSent = true;
```